src/vsr/replica.zig

```diff
@@ -6501,6 +6501,11 @@ pub fn ReplicaType(
             assert(reply_header.size > @sizeOf(Header));
             assert(destination_replica == null);
 
+            // Only allow the primary to reply, to avoid externalizing a view for which view change
+            // hasn't yet completed.
+            if (self.status != .normal) return;
+            if (!self.primary()) return;
+
```

Comment:
Ahh, so down below we aren't simply using `send_message_to_client` with the raw reply -- `send_reply_message_to_client` overrides the reply's `view` number with our own before forwarding.

I guess it took the VOPR so long to find this bug since it requires a ~simultaneous duplicate-request and a view change.

LGTM!